### DISCUSSION TITLE
Code plugin and manipulating on pasting or submitting

### DISCUSSION THREAD

**Discussion by slowrockets** [tinymce/tinymce#9321] [General]
Hello,
I am using tinymce  in vue3 with code plugin.
What I am looking for is a way to check/manipulate the content inserted into the code editor before it is submitted and stored in the main editor. I want to avoid some auto-fixing and replace it with my own. 
For example: if you paste sth like 
`<img border="0"" src="...">`
 it will result in
`&lt;img border="0""`
If you fix the "" error in border="0"" it will not escape the img  with htmlentities.
Now if I use @setContent with vue, the content will be already auto-fixed.
If I use @paste with vue or via paste_preprocess/ paste_postprocess, it will only work if I paste directly in the editor (not the code textarea)
Does anybody know a way how to access and change the content in the code plugins textarea before any auto-correction by tinymce?

**Comment by TheSpyder** [comment]
Yes, the `setContent` event happens after the content is in the editor. We have a `beforeSetContent` event as well that can be used to modify the content before it is loaded into the editor.
https://www.tiny.cloud/docs/tinymce/latest/vue-ref/#event-binding
https://www.tiny.cloud/docs/tinymce/latest/events/#editor-core-events

Note that these set content events will fire any time the `editor.setContent` API is used, not just for code view.

**Comment by slowrockets** [comment]
Hello and thanks for help!
This version indeed works. It is OK that way, I guess. However it would be really nice to bind an event on submitting the code editor content, so this parsing function (there are indeed more) would only be called when submitting code editor stuff. But I do not see any event etc available for clone plugin. I guess the only way would be to maintain our own modified version of the plugin?

**Reply by TheSpyder** [reply]
That is correct. From the editor's perspective content coming from the dialog is the same as the `setContent()` function so we don't have a unique event for the dialog.
